presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast.java

```diff
@@ -58,29 +57,23 @@ public static Block toMap(
             ConnectorSession session,
             @SqlType("map(FK,FV)") Block fromMap)
     {
-        // loop over all the parameter types and bind ConnectorSession if needed
-        // TODO: binding `ConnectorSession` should be done in function invocation framework
-        Class<?>[] parameterTypes = keyCastFunction.type().parameterArray();
-        for (int i = 0; i < parameterTypes.length; i++) {
-            if (parameterTypes[i] == ConnectorSession.class) {
-                keyCastFunction = MethodHandles.insertArguments(keyCastFunction, i, session);
-                break;
-            }
-        }
-        parameterTypes = valueCastFunction.type().parameterArray();
-        for (int i = 0; i < parameterTypes.length; i++) {
-            if (parameterTypes[i] == ConnectorSession.class) {
-                valueCastFunction = MethodHandles.insertArguments(valueCastFunction, i, session);
-                break;
-            }
-        }
+        boolean keyCastRequiresSession = keyCastFunction.type().parameterArray()[0] == ConnectorSession.class;
```

Comment:
you can inline these variables if you want as these are used only once.

Reply:
These are used in a for-loop. Now whether we want to expand them is an interesting tradeoff :)